### PR TITLE
[SPARK-37694][SQL] Disallow delete resources in spark sql cli

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

In this pr, we use spark SQL parser to process the unsupported `delete [jar|file|archive]` commands 

### Why are the changes needed?

those commands are not supported and will raise errors in all other APIs except spark-sql cli.

### Does this PR introduce _any_ user-facing change?

yes, these commands show errors instead of nothing

### How was this patch tested?

new ut

### CODE REVIEW

**Comment by yaooqinn** [comment]
cc @cloud-fan

File: sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLCLIDriver.scala [hunk 1]
```diff
@@ -371,6 +371,7 @@ private[hive] class SparkSQLCLIDriver extends CliDriver with Logging {
         // scalastyle:off println
         if (proc.isInstanceOf[Driver] || proc.isInstanceOf[SetProcessor] ||
           proc.isInstanceOf[AddResourceProcessor] || proc.isInstanceOf[ListResourceProcessor] ||
+          proc.isInstanceOf[DeleteResourceProcessor] ||
```
Comment (cloud-fan) [inline sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLCLIDriver.scala:374]: what's the code path before this PR?

Comment (yaooqinn) [reply sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLCLIDriver.scala:374]: it goes to L453 in else-branch w/ hive DeleteResourceProcessor

**Review by cloud-fan — APPROVED** [review]

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**